src/app/shared/helpers/url.ts

```diff
@@ -78,19 +78,18 @@ export function urlStringFromArray(urlAsArray: UrlCommand): string {
   }).join('');
 }
 
-function formatUrl(href: string, currentLang?: LanguageConfig): string {
-  const isAbsoluteHref = href.startsWith('http');
-  if (isAbsoluteHref) return href;
-  const url = new URL(currentLang?.path ?? '/', window.location.href);
-  url.hash = href;
-  return url.href;
+/**
+ * Convert an absolute or relative href to a full url string
+ */
+function hrefToUrl(href: string, location: Location): string {
+  return new RegExp('^(?:[a-z+]+:)?//', 'i').test(href) ? href : window.location.origin + location.prepareExternalUrl(href);
 }
```

Comment:
Href is either absolute or relative... so at the end, the function here converts to a absolute url. 

> What is the interest of using a regex here?

This matches really any absolute url. With the previous version: `HTTP://example.com` was not considered absolute while it was, `http/mypage`  was considered absolute while it was not. Here you handle external data so it is good if we are a bit more robust.   Better readability could be to do it on 2 lines with a intermediate `isAbsoute` variable.

Reply:
> Href is either absolute or relative... so at the end, the function here converts to an url (an url is absolute).
The comment above is quite bad as it should explain this just convert to something absolute by prepending the current path if href is relative.

It's just naming, so you do what you want. But no, an href is always the full string of the url, with protocol host pathname query and hash.
The only place an href might be a relative url is in the anchor href attribute, and that is because behind the scenes it populates the HTMLAnchorElement.href with the equivalent of `new URL(anchor.getAttribute('href'), window.location.href).href`.
By the way, the [URL construct signature](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) is `new URL(url [, base])` where url can be relative. So technically this is more a `urlToHref` function

> Better readability could be to do it on 2 lines with a intermediate isAbsoute variable.

I think so too, it's not much but it will be easier to read in my opinion.

Reply:
fixed